macroslib/src/cpp/rust_str.h

```diff
@@ -16,7 +16,9 @@ struct RustStrView {
     bool empty() const noexcept { return this->len == 0; }
     std::string to_std_string() const { return std::string{ data, len }; }
 #endif
-#if __cplusplus > 201402L
```

Comment:
The main check disappears 

> __cplusplus > 201402L

so if compile code with `-std=c++11` or `-std=c++14` it fails.

Reply:
Right, I'll fix it

Reply:
Meantime I've found a better way.

From the C++17 standard
```
    1 The following macro names shall be defined by the implementation:

    __cplusplus The integer literal 201703L.
```

The problem with gcc was, it defined intermediate values between 201402L and 201703L. For example in gcc 5.4 `__cplusplus` equals 201500L when run with `-std=c++17` (which, I suppose, indicates that c++17 is partially supported).

So, testing `#if __cplusplus >= 201703L` is consistent with standard and should work for gcc, clang and MSVC.